processing/dwf_calculation_algorithm.py

```
@@ -0,0 +1,396 @@
+                provider = 'spatialite'
+            )
+        )
+        
+        self.addParameter(
+            QgsProcessingParameterString(
+                'start_time',
+                self.tr('Start time of day (HH:MM:SS)'),
+                '00:00:00'
+            )
+        )
+                    
+        self.addParameter(
+            QgsProcessingParameterString(
+                'duration',
+                self.tr('Simulation duration (seconds)')
+            )
+        )
+        
+        
+        self.addParameter(
+            QgsProcessingParameterFile(
+                'dwf_progress_file',
+                self.tr('DWF progress file (.csv)'),
+                extension = 'csv',
+                defaultValue = None,
+                optional = True
+            )
+        )
+
+
+        self.addParameter(
+            QgsProcessingParameterFileDestination(
+                self.OUTPUT,
+                self.tr('Output CSV'),
+                'csv(*.csv)'
+            )
+        )
+        
+        
+
+    def processAlgorithm(self, parameters, context, feedback):
+        """
+        Here is where the processing itself takes place.
+        """
+        
+        start_time = self.parameterAsString(parameters, 'start_time', context)
+        duration = self.parameterAsDouble(parameters, 'duration', context)
+        connection_name = self.parameterAsConnectionName(parameters, self.INPUT, context)
+        output_csv = self.parameterAsFileOutput(parameters, self.OUTPUT, context)
+        dwf_factor_input = self.parameterAsFile(parameters, 'dwf_progress_file', context)
+
+        try:
+            md = QgsProviderRegistry.instance().providerMetadata('spatialite')
+            conn = md.createConnection(connection_name)
+        except QgsProviderConnectionException:
+            raise QgsProcessingException(self.tr('Could not retrieve connection details for {}').format(connection_name))
```

A connection exception can probably also mean a missing sqlite file or so? You're now providing an error message that will sometimes be wrong, I fear.

And you're hiding the original error. A `logging.exception("Error setting up connection")` line (see examples in other parts of the code) will at least log the actual error to our debug logfile.